pkg/imgpkg/bundle/images_refs.go

```diff
@@ -0,0 +1,188 @@
+// Copyright 2020 VMware, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package bundle
+
+import (
+	"fmt"
+	"strings"
+	"sync"
+
+	"github.com/google/go-containerregistry/pkg/name"
+	"github.com/google/go-containerregistry/pkg/v1/remote/transport"
+	ctlimg "github.com/k14s/imgpkg/pkg/imgpkg/image"
+	"github.com/k14s/imgpkg/pkg/imgpkg/lockconfig"
+)
+
+//go:generate go run github.com/maxbrunsfeld/counterfeiter/v6 github.com/cppforlife/go-cli-ui/ui.UI
+
+type ImageRef struct {
+	lockconfig.ImageRef
+	IsBundle *bool
+}
+
+func NewImageRef(imgRef lockconfig.ImageRef, isBundle bool) ImageRef {
+	return ImageRef{ImageRef: imgRef, IsBundle: &isBundle}
+}
+
+func (i ImageRef) DeepCopy() ImageRef {
+	return ImageRef{
+		ImageRef: i.ImageRef.DeepCopy(),
+		IsBundle: i.IsBundle,
+	}
+}
+func (i *ImageRef) updateRepo(relativeToRepo string) *ImageRef {
+	i.Image = i.replaceImageRepo(relativeToRepo)
+	return i
+}
+func (i *ImageRef) replaceImageRepo(relativeToRepo string) string {
```

Comment:
replaceImageRepo -> ImageRelativeToRepo (public). we arent really replacing anything, just calculating location. public because its used outside of ImageRef class.

Reply:
I disagree. I think we are replacing something. Specifically we are replacing the image's repository portion of its name with the argument provided.

I'm seeing this function being similar to `strings.Replace` where it takes an input and returns a copy of the input, replaced with something else.

It currently lives on the ImageRef struct. I think it could also live as a package function. And take 2 inputs. 1) an Image (string) and 2) relativeToRepo (string).

Reply:
i see, so the replacement is really related to image contents rather than related to ImageRef structure itself. my default instinct was that it's related to ImageRef since that's where it's at.

> It currently lives on the ImageRef struct. I think it could also live as a package function. And take 2 inputs. 1) an Image (string) and 2) relativeToRepo (string).

i would agree with you on this, either on ImageRefs or package.

Reply:
went with package